tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TspTool/TspTool.cs

```diff
@@ -19,11 +19,21 @@ namespace Azure.Sdk.Tools.Cli.Tools
     public class TypeSpecTool(INpxHelper npxHelper, ILogger<TypeSpecTool> logger, IOutputService output) : MCPTool
     {
 
+        // This is the template registry URL used by the TypeSpec compiler's init command.
+        private const string AzureTemplatesUrl = "https://aka.ms/typespec/azure-init";
+
         // commands
         private const string ConvertSwaggerCommandName = "convert-swagger";
+        private const string InitCommandName = "init";
 
+        // Shared command options
         private readonly Option<string> outputDirectoryArg = new("--output-directory", "The output directory for the generated TypeSpec project. This directory must already exist and be empty.") { IsRequired = true };
 
+        // Init command options
+        private readonly Option<string> templateArg = new("--template", "The template to use for the TypeSpec project. Valid values are: azure-core (for data-plane services), azure-arm (for resource-manager services)") { IsRequired = true };
+        private readonly Option<string> serviceNamespaceArg = new("--service-namespace", "The namespace of the service you are creating. This should be in Pascal case and represent the service's namespace.") { IsRequired = true };
```

Comment:
I think you can add a validation function via System.CommandLine if you want to fail fast at invocation time.